index.js

```diff
@@ -53,121 +59,142 @@ app.post('/api/upload_data', async (req, res) => {
         });
     }
 
-    let token;
-    let cpm;
-    let floor;
-    let locX;
-    let locY;
-
-    // Token verification
-    if (req.is('json')) {
-        token = req.body.token;
-        cpm = req.body.reading.cpm;
-        floor = req.body.reading.location.floor;
-        locX = req.body.reading.location.x;
-        locY = req.body.reading.location.y;
-    } else {
-        ({
-            token,
-            cpm,
-            floor,
-            locX,
-            locY,
-        } = req.body);
-    }
-
     // res.send (token + " | " + cpm + " | " + floor + " | " + locX + " | " + locY); //---DEBUG---
     // res.send (req.body); //---DEBUG---
 
     const result = await client.query({
         text: 'SELECT teamid FROM team WHERE teamtoken = $1',
-        values: [token],
+        values: [req.body.token],
     });
 
-    if (result.rowCount === 0) {
+    // Do NOT use .rowCount as that does not return number of rows in query.
+    if (result.rows.length === 0) {
         res.status(422);
-        res.send('Invalid token.');
+        res.json({
+            success: false,
+            errors: [{
+                code: 102,
+                message: 'Invalid token.',
+            }],
+        });
         return;
     }
-
     const teamID = result.rows[0].teamid;
 
     const insertQuery = 'INSERT INTO reading (teamid, posfloor, posx, posy, cpm) VALUES ($1, $2, $3, $4, $5);';
```

Comment:
Is this the query that was playing up @TimMcGilly? If so, what did you say the error was?

Reply:
@tomkettle1 Maybe, there were issues with this in the past due to wrong order of COMMIT and ROLLBACK which was fixed in previous  commit. Also changing row count here fixes a different bug. Think it works now.